commit a1c16ed2664205dd69c0ea0ec6945f7f10d3edf9
Author: Greg Kroah-Hartman <gregkh@suse.de>
Date:   Thu Oct 21 11:17:44 2010 -0700

    Staging: brcm80211: remove typedefs.h
    
    It's not needed anymore, so remove it and clean up the .h inclusion mess
    a bit to get everything to build properly again.
    
    Wow, this needs some unwinding...
    
    Cc: Brett Rudley <brudley@broadcom.com>
    Cc: Henry Ptasinski <henryp@broadcom.com>
    Cc: Nohee Ko <noheek@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh.c
@@ -15,7 +15,8 @@
  */
 /* ****************** BCMSDH Interface Functions *************************** */
 
-#include <typedefs.h>
+#include <linux/types.h>
+#include <bcmdefs.h>
 #include <bcmdevs.h>
 #include <bcmendian.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh_linux.c
@@ -20,7 +20,6 @@
 
 #define __UNDEF_NO_VERSION__
 
-#include <typedefs.h>
 #include <linuxver.h>
 
 #include <linux/pci.h>
--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh_sdmmc.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh_sdmmc.c
@@ -13,8 +13,8 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-#include <typedefs.h>
-
+#include <linux/types.h>
+#include <bcmdefs.h>
 #include <bcmdevs.h>
 #include <bcmendian.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh_sdmmc_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh_sdmmc_linux.c
@@ -13,15 +13,14 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
-#include <typedefs.h>
+#include <linux/types.h>
+#include <linux/sched.h>	/* request_irq() */
+#include <bcmdefs.h>
 #include <bcmutils.h>
 #include <sdio.h>		/* SDIO Specs */
 #include <bcmsdbus.h>		/* bcmsdh to/from specific controller APIs */
 #include <sdiovar.h>		/* to get msglevel bit values */
 
-#include <linux/sched.h>	/* request_irq() */
-
 #include <linux/mmc/core.h>
 #include <linux/mmc/card.h>
 #include <linux/mmc/sdio_func.h>
--- a/drivers/staging/brcm80211/brcmfmac/dhd_cdc.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_cdc.c
@@ -14,7 +14,8 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
+#include <linux/types.h>
+#include <bcmdefs.h>
 #include <osl.h>
 
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/brcmfmac/dhd_common.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_common.c
@@ -13,10 +13,10 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-#include <typedefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <bcmutils.h>
 #include <bcmendian.h>
 #include <dngl_stats.h>
--- a/drivers/staging/brcm80211/brcmfmac/dhd_custom_gpio.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_custom_gpio.c
@@ -14,7 +14,6 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
 #include <linuxver.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/brcmfmac/dhd_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_linux.c
@@ -17,10 +17,6 @@
 #ifdef CONFIG_WIFI_CONTROL_FUNC
 #include <linux/platform_device.h>
 #endif
-#include <typedefs.h>
-#include <linuxver.h>
-#include <osl.h>
-
 #include <linux/init.h>
 #include <linux/kernel.h>
 #include <linux/kthread.h>
@@ -34,8 +30,10 @@
 #include <linux/ethtool.h>
 #include <linux/fcntl.h>
 #include <linux/fs.h>
-
 #include <linux/uaccess.h>
+#include <bcmdefs.h>
+#include <linuxver.h>
+#include <osl.h>
 #include <bcmutils.h>
 #include <bcmendian.h>
 
--- a/drivers/staging/brcm80211/brcmfmac/dhd_sdio.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_sdio.c
@@ -14,7 +14,8 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
+#include <linux/types.h>
+#include <bcmdefs.h>
 #include <osl.h>
 #include <bcmsdh.h>
 
--- a/drivers/staging/brcm80211/brcmfmac/wl_cfg80211.c
+++ b/drivers/staging/brcm80211/brcmfmac/wl_cfg80211.c
@@ -14,16 +14,15 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
+#include <linux/kernel.h>
+#include <linux/if_arp.h>
 #include <linuxver.h>
 #include <osl.h>
-#include <linux/kernel.h>
 
 #include <bcmutils.h>
 #include <bcmendian.h>
 #include <proto/ethernet.h>
 
-#include <linux/if_arp.h>
 #include <asm/uaccess.h>
 
 #include <dngl_stats.h>
@@ -35,7 +34,6 @@
 #include <dngl_stats.h>
 #include <dhd.h>
 
-#include <linux/kernel.h>
 #include <linux/kthread.h>
 #include <linux/netdevice.h>
 #include <linux/sched.h>
--- a/drivers/staging/brcm80211/brcmfmac/wl_cfg80211.h
+++ b/drivers/staging/brcm80211/brcmfmac/wl_cfg80211.h
@@ -18,11 +18,10 @@
 #define _wl_cfg80211_h_
 
 #include <linux/wireless.h>
-#include <typedefs.h>
-#include <proto/ethernet.h>
-#include <wlioctl.h>
 #include <linux/wireless.h>
 #include <net/cfg80211.h>
+#include <proto/ethernet.h>
+#include <wlioctl.h>
 
 struct wl_conf;
 struct wl_iface;
--- a/drivers/staging/brcm80211/brcmfmac/wl_iw.c
+++ b/drivers/staging/brcm80211/brcmfmac/wl_iw.c
@@ -14,12 +14,11 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <wlioctl.h>
-
-#include <typedefs.h>
-#include <linuxver.h>
 #include <linux/kthread.h>
+#include <bcmdefs.h>
+#include <linuxver.h>
 #include <osl.h>
+#include <wlioctl.h>
 
 #include <bcmutils.h>
 #include <bcmendian.h>
--- a/drivers/staging/brcm80211/brcmfmac/wl_iw.h
+++ b/drivers/staging/brcm80211/brcmfmac/wl_iw.h
@@ -19,7 +19,6 @@
 
 #include <linux/wireless.h>
 
-#include <typedefs.h>
 #include <proto/ethernet.h>
 #include <wlioctl.h>
 
--- a/drivers/staging/brcm80211/include/bcm_rpc.h
+++ b/drivers/staging/brcm80211/include/bcm_rpc.h
@@ -17,7 +17,6 @@
 #ifndef _BCM_RPC_H_
 #define _BCM_RPC_H_
 
-#include <typedefs.h>
 #include <rpc_osl.h>
 
 typedef struct rpc_info rpc_info_t;
--- a/drivers/staging/brcm80211/include/bcmdefs.h
+++ b/drivers/staging/brcm80211/include/bcmdefs.h
@@ -24,10 +24,17 @@
 #define JTAG_BUS		4
 #define USB_BUS			5
 #define SPI_BUS			6
-/*
- * One doesn't need to include this file explicitly, gets included automatically if
- * typedefs.h is included.
- */
+
+
+#ifndef OFF
+#define	OFF	0
+#endif
+
+#ifndef ON
+#define	ON	1		/* ON = 1 */
+#endif
+
+#define	AUTO	(-1)		/* Auto = -1 */
 
 #ifdef mips
 #define BCMFASTPATH		__attribute__ ((__section__(".text.fastpath")))
--- a/drivers/staging/brcm80211/include/bcmendian.h
+++ b/drivers/staging/brcm80211/include/bcmendian.h
@@ -17,8 +17,6 @@
 #ifndef _BCMENDIAN_H_
 #define _BCMENDIAN_H_
 
-#include <typedefs.h>
-
 /* Reverse the bytes in a 16-bit value */
 #define BCMSWAP16(val) \
 	((u16)((((u16)(val) & (u16)0x00ffU) << 8) | \
--- a/drivers/staging/brcm80211/include/bcmnvram.h
+++ b/drivers/staging/brcm80211/include/bcmnvram.h
@@ -19,7 +19,6 @@
 
 #ifndef _LANGUAGE_ASSEMBLY
 
-#include <typedefs.h>
 #include <bcmdefs.h>
 
 struct nvram_header {
--- a/drivers/staging/brcm80211/include/d11.h
+++ b/drivers/staging/brcm80211/include/d11.h
@@ -17,7 +17,7 @@
 #ifndef	_D11_H
 #define	_D11_H
 
-#include <typedefs.h>
+#include <bcmdefs.h>
 #include <bcmdevs.h>
 #include <hndsoc.h>
 #include <sbhndpio.h>
--- a/drivers/staging/brcm80211/include/dbus.h
+++ b/drivers/staging/brcm80211/include/dbus.h
@@ -17,8 +17,6 @@
 #ifndef __DBUS_H__
 #define __DBUS_H__
 
-#include "typedefs.h"
-
 #ifdef BCMDBG
 #define DBUSERR(args)        do { if (net_ratelimit()) printf args; } while (0)
 #define DBUSTRACE(args)
--- a/drivers/staging/brcm80211/include/dhdioctl.h
+++ b/drivers/staging/brcm80211/include/dhdioctl.h
@@ -17,8 +17,6 @@
 #ifndef _dhdioctl_h_
 #define	_dhdioctl_h_
 
-#include <typedefs.h>
-
 /* require default structure packing */
 #define BWL_DEFAULT_PACKING
 #include <packed_section_start.h>
--- a/drivers/staging/brcm80211/include/hndrte_armtrap.h
+++ b/drivers/staging/brcm80211/include/hndrte_armtrap.h
@@ -47,8 +47,6 @@
 
 #ifndef	_LANGUAGE_ASSEMBLY
 
-#include <typedefs.h>
-
 typedef struct _trap_struct {
 	u32 type;
 	u32 epc;
--- a/drivers/staging/brcm80211/include/hndrte_cons.h
+++ b/drivers/staging/brcm80211/include/hndrte_cons.h
@@ -14,8 +14,6 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
-
 #define CBUF_LEN	(128)
 
 #define LOG_BUF_LEN	1024
--- a/drivers/staging/brcm80211/include/linux_osl.h
+++ b/drivers/staging/brcm80211/include/linux_osl.h
@@ -17,9 +17,6 @@
 #ifndef _linux_osl_h_
 #define _linux_osl_h_
 
-#ifdef BRCM_FULLMAC
-#include <typedefs.h>
-#endif
 
 /* Linux Kernel: File Operations: start */
 extern void *osl_os_open_image(char *filename);
--- a/drivers/staging/brcm80211/include/msgtrace.h
+++ b/drivers/staging/brcm80211/include/msgtrace.h
@@ -17,8 +17,6 @@
 #ifndef	_MSGTRACE_H
 #define	_MSGTRACE_H
 
-#include <typedefs.h>
-
 /* This marks the start of a packed structure section. */
 #include <packed_section_start.h>
 
--- a/drivers/staging/brcm80211/include/proto/802.11.h
+++ b/drivers/staging/brcm80211/include/proto/802.11.h
@@ -17,7 +17,6 @@
 #ifndef _802_11_H_
 #define _802_11_H_
 
-#include <typedefs.h>
 #include <proto/wpa.h>
 #include <packed_section_start.h>
 
--- a/drivers/staging/brcm80211/include/proto/bcmeth.h
+++ b/drivers/staging/brcm80211/include/proto/bcmeth.h
@@ -17,7 +17,6 @@
 #ifndef _BCMETH_H_
 #define _BCMETH_H_
 
-#include <typedefs.h>
 #include <packed_section_start.h>
 
 #define	BCMILCP_SUBTYPE_RATE		1
--- a/drivers/staging/brcm80211/include/proto/bcmevent.h
+++ b/drivers/staging/brcm80211/include/proto/bcmevent.h
@@ -17,7 +17,6 @@
 #ifndef _BCMEVENT_H_
 #define _BCMEVENT_H_
 
-#include <typedefs.h>
 #include <packed_section_start.h>
 
 #define BCM_EVENT_MSG_VERSION		1
--- a/drivers/staging/brcm80211/include/proto/bcmip.h
+++ b/drivers/staging/brcm80211/include/proto/bcmip.h
@@ -17,7 +17,6 @@
 #ifndef _bcmip_h_
 #define _bcmip_h_
 
-#include <typedefs.h>
 #include <packed_section_start.h>
 
 #define IP_VER_OFFSET		0x0
--- a/drivers/staging/brcm80211/include/proto/ethernet.h
+++ b/drivers/staging/brcm80211/include/proto/ethernet.h
@@ -17,7 +17,6 @@
 #ifndef _NET_ETHERNET_H_
 #define _NET_ETHERNET_H_
 
-#include "typedefs.h"
 #include <packed_section_start.h>
 
 #define	ETHER_ADDR_LEN		6
--- a/drivers/staging/brcm80211/include/proto/vlan.h
+++ b/drivers/staging/brcm80211/include/proto/vlan.h
@@ -17,7 +17,6 @@
 #ifndef _vlan_h_
 #define _vlan_h_
 
-#include <typedefs.h>
 #include <packed_section_start.h>
 
 #define VLAN_VID_MASK		0xfff
--- a/drivers/staging/brcm80211/include/proto/wpa.h
+++ b/drivers/staging/brcm80211/include/proto/wpa.h
@@ -17,7 +17,6 @@
 #ifndef _proto_wpa_h_
 #define _proto_wpa_h_
 
-#include <typedefs.h>
 #include <proto/ethernet.h>
 
 #include <packed_section_start.h>
--- a/drivers/staging/brcm80211/include/qmath.h
+++ b/drivers/staging/brcm80211/include/qmath.h
@@ -17,8 +17,6 @@
 #ifndef __QMATH_H__
 #define __QMATH_H__
 
-#include <typedefs.h>
-
 s16 qm_sat32(s32 op);
 
 s32 qm_mul321616(s16 op1, s16 op2);
--- a/drivers/staging/brcm80211/include/sdiovar.h
+++ b/drivers/staging/brcm80211/include/sdiovar.h
@@ -17,8 +17,6 @@
 #ifndef _sdiovar_h_
 #define _sdiovar_h_
 
-#include <typedefs.h>
-
 /* require default structure packing */
 #define BWL_DEFAULT_PACKING
 #include <packed_section_start.h>
--- a/drivers/staging/brcm80211/include/typedefs.h
+++ /dev/null
@@ -1,39 +0,0 @@
-/*
- * Copyright (c) 2010 Broadcom Corporation
- *
- * Permission to use, copy, modify, and/or distribute this software for any
- * purpose with or without fee is hereby granted, provided that the above
- * copyright notice and this permission notice appear in all copies.
- *
- * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
- * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
- * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
- * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
- * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
- * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
- * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
- */
-
-#ifndef _TYPEDEFS_H_
-#define _TYPEDEFS_H_
-
-#include <linux/types.h>
-
-#ifndef OFF
-#define	OFF	0
-#endif
-
-#ifndef ON
-#define	ON	1		/* ON = 1 */
-#endif
-
-#define	AUTO	(-1)		/* Auto = -1 */
-
-
-/*
- * Including the bcmdefs.h here, to make sure everyone including typedefs.h
- * gets this automatically
-*/
-#include <bcmdefs.h>
-
-#endif				/* _TYPEDEFS_H_ */
--- a/drivers/staging/brcm80211/include/wlioctl.h
+++ b/drivers/staging/brcm80211/include/wlioctl.h
@@ -17,7 +17,6 @@
 #ifndef _wlioctl_h_
 #define	_wlioctl_h_
 
-#include <typedefs.h>
 #include <proto/ethernet.h>
 #ifdef BRCM_FULLMAC
 #include <proto/bcmeth.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_cmn.c
+++ b/drivers/staging/brcm80211/phy/wlc_phy_cmn.c
@@ -16,10 +16,10 @@
 
 #include <wlc_cfg.h>
 
-#include <typedefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <bcmendian.h>
 #include <bcmnvram.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_hal.h
+++ b/drivers/staging/brcm80211/phy/wlc_phy_hal.h
@@ -17,7 +17,6 @@
 #ifndef _wlc_phy_h_
 #define _wlc_phy_h_
 
-#include <typedefs.h>
 #include <wlioctl.h>
 #include <siutils.h>
 #include <d11.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_int.h
+++ b/drivers/staging/brcm80211/phy/wlc_phy_int.h
@@ -17,7 +17,8 @@
 #ifndef _wlc_phy_int_h_
 #define _wlc_phy_int_h_
 
-#include <typedefs.h>
+#include <linux/kernel.h>
+#include <bcmdefs.h>
 #include <bcmutils.h>
 
 #include <bcmsrom_fmt.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_lcn.c
+++ b/drivers/staging/brcm80211/phy/wlc_phy_lcn.c
@@ -14,14 +14,14 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linux/bitops.h>
 #include <wlc_cfg.h>
 #include <qmath.h>
 #include <osl.h>
-#include <linux/kernel.h>
-#include <linux/string.h>
 #include <linuxver.h>
 #include <siutils.h>
-#include <linux/bitops.h>
 #include <hndpmu.h>
 
 #include <wlc_phy_radio.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_lcn.h
+++ b/drivers/staging/brcm80211/phy/wlc_phy_lcn.h
@@ -17,8 +17,6 @@
 #ifndef _wlc_phy_lcn_h_
 #define _wlc_phy_lcn_h_
 
-#include <typedefs.h>
-
 struct phy_info_lcnphy {
 	int lcnphy_txrf_sp_9_override;
 	u8 lcnphy_full_cal_channel;
--- a/drivers/staging/brcm80211/phy/wlc_phy_n.c
+++ b/drivers/staging/brcm80211/phy/wlc_phy_n.c
@@ -14,10 +14,10 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <wlc_cfg.h>
-#include <typedefs.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <wlc_cfg.h>
 #include <linuxver.h>
 #include <osl.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/phy/wlc_phytbl_lcn.c
+++ b/drivers/staging/brcm80211/phy/wlc_phytbl_lcn.c
@@ -14,8 +14,7 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
-
+#include <linux/types.h>
 #include <wlc_phy_int.h>
 #include <wlc_phytbl_lcn.h>
 
--- a/drivers/staging/brcm80211/phy/wlc_phytbl_n.c
+++ b/drivers/staging/brcm80211/phy/wlc_phytbl_n.c
@@ -14,7 +14,7 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
+#include <linux/kernel.h>
 
 #include <wlc_phy_int.h>
 #include <wlc_phytbl_n.h>
--- a/drivers/staging/brcm80211/sys/wl_mac80211.c
+++ b/drivers/staging/brcm80211/sys/wl_mac80211.c
@@ -16,13 +16,13 @@
 
 #define __UNDEF_NO_VERSION__
 
-#include <typedefs.h>
-#include <linuxver.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/etherdevice.h>
 #include <linux/string.h>
 #include <linux/pci_ids.h>
+#include <bcmdefs.h>
+#include <linuxver.h>
+#include <osl.h>
 #define WLC_MAXBSSCFG		1	/* single BSS configs */
 
 #include <wlc_cfg.h>
--- a/drivers/staging/brcm80211/sys/wl_ucode.h
+++ b/drivers/staging/brcm80211/sys/wl_ucode.h
@@ -14,7 +14,6 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
 typedef struct d11init {
 	u16 addr;
 	u16 size;
--- a/drivers/staging/brcm80211/sys/wl_ucode_loader.c
+++ b/drivers/staging/brcm80211/sys/wl_ucode_loader.c
@@ -15,7 +15,8 @@
  */
 
 typedef struct wl_info wl_info_t;
-#include <typedefs.h>
+#include <linux/types.h>
+#include <bcmdefs.h>
 #include <d11ucode_ext.h>
 #include <wl_ucode.h>
 
--- a/drivers/staging/brcm80211/sys/wlc_alloc.c
+++ b/drivers/staging/brcm80211/sys/wlc_alloc.c
@@ -13,11 +13,10 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
-#include <wlc_cfg.h>
-#include <typedefs.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <wlc_cfg.h>
 #include <linuxver.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_ampdu.c
+++ b/drivers/staging/brcm80211/sys/wlc_ampdu.c
@@ -13,9 +13,9 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
+#include <linux/kernel.h>
 #include <wlc_cfg.h>
-#include <typedefs.h>
+#include <bcmdefs.h>
 #include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
--- a/drivers/staging/brcm80211/sys/wlc_antsel.c
+++ b/drivers/staging/brcm80211/sys/wlc_antsel.c
@@ -18,7 +18,6 @@
 
 #ifdef WLANTSEL
 
-#include <typedefs.h>
 #include <linux/kernel.h>
 #include <linuxver.h>
 #include <bcmdefs.h>
--- a/drivers/staging/brcm80211/sys/wlc_bmac.c
+++ b/drivers/staging/brcm80211/sys/wlc_bmac.c
@@ -18,12 +18,11 @@
 #error "This file needs WLC_LOW"
 #endif
 
+#include <linux/kernel.h>
 #include <wlc_cfg.h>
-#include <typedefs.h>
 #include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
-#include <linux/kernel.h>
 #include <proto/802.11.h>
 #include <bcmwifi.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_channel.c
+++ b/drivers/staging/brcm80211/sys/wlc_channel.c
@@ -14,12 +14,11 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <wlc_cfg.h>
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <wlc_cfg.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_event.c
+++ b/drivers/staging/brcm80211/sys/wlc_event.c
@@ -14,8 +14,8 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
 #include <linux/kernel.h>
+#include <bcmdefs.h>
 #include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_mac80211.c
+++ b/drivers/staging/brcm80211/sys/wlc_mac80211.c
@@ -13,14 +13,12 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
-#include <wlc_cfg.h>
-#include <typedefs.h>
 #include <linux/kernel.h>
-#include <linuxver.h>
+#include <linux/ctype.h>
 #include <bcmdefs.h>
+#include <wlc_cfg.h>
+#include <linuxver.h>
 #include <osl.h>
-#include <linux/ctype.h>
 #include <bcmutils.h>
 #include <bcmwifi.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_phy_shim.c
+++ b/drivers/staging/brcm80211/sys/wlc_phy_shim.c
@@ -21,12 +21,11 @@
  *   access to wlc_hw pointer.
  */
 
+#include <linux/kernel.h>
+#include <bcmdefs.h>
 #include <wlc_cfg.h>
-#include <typedefs.h>
 #include <linuxver.h>
-#include <linux/kernel.h>
 #include <bcmutils.h>
-#include <bcmdefs.h>
 #include <osl.h>
 
 #include <proto/802.11.h>
--- a/drivers/staging/brcm80211/sys/wlc_rate.c
+++ b/drivers/staging/brcm80211/sys/wlc_rate.c
@@ -13,11 +13,10 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
+#include <linux/kernel.h>
+#include <bcmdefs.h>
 #include <wlc_cfg.h>
-#include <typedefs.h>
 #include <osl.h>
-#include <linux/kernel.h>
 #include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_stf.c
+++ b/drivers/staging/brcm80211/sys/wlc_stf.c
@@ -14,9 +14,8 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <wlc_cfg.h>
-#include <typedefs.h>
 #include <linux/kernel.h>
+#include <wlc_cfg.h>
 #include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
--- a/drivers/staging/brcm80211/util/aiutils.c
+++ b/drivers/staging/brcm80211/util/aiutils.c
@@ -14,11 +14,10 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/util/bcmotp.c
+++ b/drivers/staging/brcm80211/util/bcmotp.c
@@ -14,11 +14,10 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <bcmdevs.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/util/bcmsrom.c
+++ b/drivers/staging/brcm80211/util/bcmsrom.c
@@ -13,12 +13,10 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <stdarg.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/util/bcmutils.c
+++ b/drivers/staging/brcm80211/util/bcmutils.c
@@ -14,13 +14,12 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <stdarg.h>
-#include <osl.h>
 #include <linux/ctype.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <stdarg.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/util/bcmwifi.c
+++ b/drivers/staging/brcm80211/util/bcmwifi.c
@@ -13,12 +13,9 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
-#include <typedefs.h>
-
-#include <osl.h>
 #include <linux/ctype.h>
 #include <linux/kernel.h>
+#include <bcmdefs.h>
 #include <bcmutils.h>
 #include <bcmwifi.h>
 
--- a/drivers/staging/brcm80211/util/hnddma.c
+++ b/drivers/staging/brcm80211/util/hnddma.c
@@ -14,7 +14,6 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
 #include <linuxver.h>
--- a/drivers/staging/brcm80211/util/hndpmu.c
+++ b/drivers/staging/brcm80211/util/hndpmu.c
@@ -13,13 +13,11 @@
  * OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
-
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <osl.h>
+#include <linux/kernel.h>
 #include <linux/string.h>
 #include <linuxver.h>
-#include <linux/kernel.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <bcmdevs.h>
--- a/drivers/staging/brcm80211/util/linux_osl.c
+++ b/drivers/staging/brcm80211/util/linux_osl.c
@@ -14,19 +14,18 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
+#include <linux/delay.h>
+#include <linux/fs.h>
+#ifdef mips
+#include <asm/paccess.h>
+#endif				/* mips */
 #include <bcmendian.h>
 #include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
-#include <linux/delay.h>
-#ifdef mips
-#include <asm/paccess.h>
-#endif				/* mips */
 #include <pcicfg.h>
 
-#include <linux/fs.h>
 
 #define PCI_CFG_RETRY 		10
 
--- a/drivers/staging/brcm80211/util/nicpci.c
+++ b/drivers/staging/brcm80211/util/nicpci.c
@@ -14,7 +14,6 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
 #include <linux/string.h>
 #include <linuxver.h>
 #include <bcmdefs.h>
--- a/drivers/staging/brcm80211/util/nvram/nvram_ro.c
+++ b/drivers/staging/brcm80211/util/nvram/nvram_ro.c
@@ -15,9 +15,8 @@
  */
 
 #include <linux/slab.h>
-#include <typedefs.h>
-#include <bcmdefs.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/util/qmath.c
+++ b/drivers/staging/brcm80211/util/qmath.c
@@ -14,6 +14,7 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
+#include <linux/types.h>
 #include "qmath.h"
 
 /*
--- a/drivers/staging/brcm80211/util/sbutils.c
+++ b/drivers/staging/brcm80211/util/sbutils.c
@@ -14,7 +14,7 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
+#include <linux/types.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/util/siutils.c
+++ b/drivers/staging/brcm80211/util/siutils.c
@@ -14,11 +14,10 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
-#include <typedefs.h>
-#include <bcmdefs.h>
-#include <osl.h>
 #include <linux/kernel.h>
 #include <linux/string.h>
+#include <bcmdefs.h>
+#include <osl.h>
 #include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
